library/std/src/sys/common/alloc.rs

```diff
@@ -14,7 +14,8 @@ use crate::ptr;
     target_arch = "asmjs",
     target_arch = "wasm32",
     target_arch = "hexagon",
-    target_arch = "riscv32"
+    target_arch = "riscv32",
+    target_arch = "xtensa"
```

Comment:
I think xtensa is not supported yet.

Reply:
Support for Xtensa touches multiple crates and does not necessarily land into all of them in an atomic fashion:
- (1) `libc` crate: Here, it landed for the first time [> 1 year ago](https://github.com/rust-lang/libc/pull/1748) and was [recently extended](https://github.com/rust-lang/libc/pull/2310) with extra APIs and reworked to not really mean "Xtensa" but ESP-IDF for Risv32 *and* Xtensa and any architecture supported by ESP-IDf in fact.
- (2) `library/std`: Here, it lands now. The only place that needs "Xtensa" explicitly listed (as opposed to branching based on `target_os = "espidf"`) is this one AFAIR.
- (3) `rust/compiler/rustc_target`: Not landed yet, but we have a patchset for that. However, the Xtensa arch needs to be upstreamed in LLVM first.

Now, if you or somebody else feels uncomfortable having this conditional before the actual support for the Xtensa *targets* being upstreamed first, I can remove that line.

But if it does not do harm, why bother? We'll have to add it back anyway once (3) lands. And in the meantime, we'll know that libStd is Xtensa "ready" once the Xtensa targets get upstreamed.